lib/graphql/transfer.ts

```diff
@@ -203,7 +212,41 @@ export class Transfer extends Model<CreateTransferInput> {
     return result.confirmCancelTransfer;
   }
 
-  async fetch(args?: FetchOptions): Promise<ResultPage<TransferEntry>> {
-    throw new Error("not implemented yet");
+  /**
+   * @override specify query parameter arguments for transfers
+   *
+   * @param args query parameters
+   */
+  createAsyncIterator(args: TransferFetchOptions) {
+    return super.createAsyncIterator(args);
+  }
+
+  /**
+   * @override specify query parameter arguments for transfers
+   *
+   * @param args query parameters
+   */
+  fetchAll(args: TransferFetchOptions) {
```

Comment:
Is it fine to override the base `IterableModel` methods here just to provide the correct argument type? I don't know if there is a better typescript way to do this

Reply:
Is it possible to use generics here? Sth. like IterableModel<T>

Reply:
Great idea, I gave it a shot in this commit: f9b3bc212d90ec8b70184cc7a7f8ff68c8ab4f4f

It is fine for the most part but I have a question, I was forced to cast like so in a couple of places:
https://github.com/kontist/js-sdk/blob/f9b3bc212d90ec8b70184cc7a7f8ff68c8ab4f4f/lib/graphql/iterableModel.ts#L21-L24
otherwise I get an infamous `... is assignable to the constraint of type 'FetchOptionsType', but 'FetchOptionsType' could be instantiated with a different subtype of constraint` which makes sense in a way but I could not find another way around it

Reply:
see https://github.com/kontist/js-sdk/pull/36#discussion_r350395494

TypeScript is right here, but message is not very comprehensible imho: Reason for this seems that `args` is allowed to be optional and then `{ ...args, after: lastResult.pageInfo.endCursor }`no longer satisfies `FetchOptionsType` (but has to because of `model: IFetch<ModelType, FetchOptionsType>,`) which could be set to e.g. `{ mandatoryProperty: string }` which would be missing then.
Hope this makes sense ;-) Let us discuss if you want to.

Reply:
Hmm it does absolutely make sense yes, but if I make `args` mandatory, I have to make it all the way up to `fetchAll(args: FetchOptionsType)`.

However for transactions, `args` has no mandatory field, so this would force users to go
`for await (const transaction of client.models.transaction.fetchAll({}))` (empty object argument) which is not so great 🤔 Or maybe is there a way to prevent this that I don't see? (apart from overriding `fetchAll` in transaction model, I don't see it 🙁)

Reply:
An illustration of what I mean can be seen in this commit: 2a704ffd76842829bfccfa39a09ca1a6c5ff663f 🙂